tests/seed/catalog/test_product.py

```diff
@@ -42,19 +41,18 @@ async def test_get_product_without_id(context: Context) -> None:
 
 
 async def test_get_or_create_product_create_new(
-    context: Context, vendor_client, products_service, product
+    context: Context, vendor_client, products_service, product, fake_file_factory
 ) -> None:
     products_service.create.return_value = product
     vendor_client.catalog.products = products_service
-    fake_icon_bytes = io.BytesIO(b"fake image")
-
     with (
-        patch("seed.catalog.product.get_product", return_value=None),
+        patch("seed.catalog.product.get_product", new_callable=AsyncMock) as mock_get_product,
         patch("seed.catalog.product.icon", new=MagicMock()),
-        patch("pathlib.Path.open", return_value=fake_icon_bytes),
+        patch("pathlib.Path.open", new=lambda *_, **_unused: fake_file_factory()),
```

Comment:
It may be an overkill, but why don't we start using https://pytest-pyfakefs.readthedocs.io/ for files? It would give us certain things out of the box?